prov/gni/src/gnix_msg.c

```diff
@@ -183,14 +185,24 @@ int _gnix_ep_eager_msg_w_data_match(struct gnix_fid_ep *ep, void *msg,
 	 * or we add the request to the tail of the unexpected queue
 	 */
 
+	if (sflags & GNIX_MSG_TAGGED) {
```

Comment:
might be a good case for using ```likely```.  I think the tagged interface will be used much more for high performance MPI than the non-matching send/recv interface.